### PR TITLE
Feature/13 user settings back

### PR DESCRIPTION
**Pull request by f3rland**
### What it does

- Fix linting settings for multi-platform dev environments
- Lean User object return
- `email` and `terms` fields added to User model (see `GET /auth/me`)
- `GET & PATCH /auth/me` unit tests
- async unit tests
- Middleware to validate authentication and parameters

### What to look for

- Working endpoints
- @splantebulldozer , see `server/requests.http` for endpoints reference

### What to avoid

- Unit tests code can be simplified

### CODE REVIEW

**Review by splantebulldozer — APPROVED** [review]

File: server/controllers/auth.js [hunk 1]
```diff
@@ -36,7 +31,9 @@ exports.SignUp = async function test(req, res, next) {
   const { password, companyName } = req.body;
 
   // Create document
-  const user = new User({ username, companyName, password });
+  const user = new User({
+    username, companyName, password, email: username, terms: [companyName],
```
Comment (skassam21) [inline server/controllers/auth.js:35]: 👍

File: server/models/user.js [hunk 2]
```diff
@@ -20,11 +20,21 @@ const User = new Schema({
...
+  terms: {
+    type: [String],
+    required: false,
+  },
+  email: {
+    type: String,
+    required: false,
+  },
+  crawlers: {
+    type: [String],
```
Comment (skassam21) [inline server/models/user.js:32]: Are these basically the specific crawlers that are active? In that case, let's add a enum of potential crawlers that can be added (i.e. twitter, reddit, etc.)

File: server/controllers/auth.js [hunk 3]
```diff
@@ -47,9 +44,14 @@ exports.Login = async (req, res, next) => {
...
-exports.Self = async (req, res) => {
-  if (!req.user) {
-    return res.status(401).send();
-  }
-  return res.status(200).send(req.user);
+exports.Self = async (req, res) => res.status(200).send(req.user);
+
+exports.UpdateProfile = async (req, res) => {
+  const { terms, email, crawlers } = req.body;
+  const obj = { terms, email, crawlers };
```
Comment (skassam21) [inline server/controllers/auth.js]: we should remove duplicates in terms right?

**Review by skassam21 — COMMENTED** [review]
Couple of thoughts, looks good though

File: server/controllers/auth.js (hunk 3 above)
Comment (f3rland) [reply server/controllers/auth.js]: Without giving insight to the user/front? 🤔 
I think we should warn the user about altering it's data.
I've added validation as a safety, but should be validated front first.

I've also removed duplicates from crawlers.

File: server/models/user.js (hunk 2 above)
Comment (f3rland) [reply server/models/user.js:32]: Will be handled in the CrawlerService later.
A method that expose available crawlers will be used to sanitize incoming data through a custom validator (see https://mongoosejs.com/docs/validation.html#custom-validators)

**Review by f3rland — COMMENTED** [review]
Recommendations applied

**Review by skassam21 — APPROVED** [review]

**Pull request merged**